label_studio_converter/converter.py

```diff
@@ -223,52 +224,65 @@ def iter_from_dir(self, input_dir):
                            'where value is list of dicts')
-        result = []
-        if has_annotations:
-            # get last not skipped completion and make result from it
-            annotations = d.get('annotations')
-            if annotations is None:
-                annotations = d['completions']
-            tmp = list(filter(lambda x: not (x.get('skipped', False) or x.get('was_cancelled', False)), annotations))
-            if len(tmp) > 0:
-                # TODO: only one annotation per task is supported for non full JSON formats
-                result = sorted(tmp, key=lambda x: x.get('created_at', 0), reverse=True)[0]['result']
-            else:
-                return None
-
-        elif 'result' in d:
-            result = d['result']
-        inputs = d['data']
-        outputs = defaultdict(list)
-        for r in result:
-            if 'from_name' in r and r['from_name'] in self._output_tags:
-                v = deepcopy(r['value'])
-                v['type'] = self._schema[r['from_name']]['type']
-                if 'original_width' in r:
-                    v['original_width'] = r['original_width']
-                if 'original_height' in r:
-                    v['original_height'] = r['original_height']
-                outputs[r['from_name']].append(v)
-        return {
-            'id': d['id'],
-            'input': inputs,
-            'output': outputs
-        }
+
+        # get last not skipped completion and make result from it
+        annotations = task['annotations'] if 'annotations' in task else task['completions']
+        
+        # skip cancelled annotations
+        cancelled = lambda x: not (x.get('skipped', False) or x.get('was_cancelled', False))
+        annotations = list(filter(cancelled, annotations))
+        if not annotations:
+            return None
+        
+        # sort by creation time
+        annotations = sorted(annotations, key=lambda x: x.get('created_at', 0), reverse=True)
+
+        for annotation in annotations:
+            inputs = task['data']
+            result = annotation['result']
+            outputs = defaultdict(list)
+
+            # get results only as output
+            for r in result:
+                if 'from_name' in r and r['from_name'] in self._output_tags:
+                    v = deepcopy(r['value'])
+                    v['type'] = self._schema[r['from_name']]['type']
+                    if 'original_width' in r:
+                        v['original_width'] = r['original_width']
+                    if 'original_height' in r:
+                        v['original_height'] = r['original_height']
+                    outputs[r['from_name']].append(v)
+
+            yield {
+                'id': task['id'],
+                'input': inputs,
+                'output': outputs,
+                'completed_by': annotation.get('completed_by', {}),
+                'annotation_id': annotation.get('id')
```

Comment:
why we need `annotation_id` ?

Reply:
To understand what annotation exactly should I edit if it has errors

Reply:
but (id, completed_by) also defines that uniquely

Reply:
It will be convenient for annotation grid view / annotation list. And now we have no restrictions to create multiple annotations by one user